tests/integration/cattletest/core/test_svc_discovery.py

```diff
@@ -1457,10 +1457,10 @@ def test_global_service_update_label(new_context):
     instance2_host = instance2.hosts()[0].id
     assert instance2_host == host2.id
 
-    # destroy the instance, reactivate the service and check
-    # both hosts got instances
-    _instance_remove(instance1, client)
+    # deactivate the service, destroy the instance,
+    # reactivate the service, and check both hosts got instances
     service = wait_state(client, service.deactivate(), 'inactive')
+    _instance_remove(instance1, client)
```

Comment:
@alena1108  The failure in this test (where it just times out waiting) is due to a bug that @ibuildthecloud wants to fix for 1.4.0.

Heres's the problem:
When moving a resource from a transitioning state to a resting state, we don't check what the transitioning state is.

So, the instance is in stopping, this code submits  a DELETE. The API call succeeeds and resource is moved to removing. BUT then stop finishes and just stomps on the state and moves it from removing to stopped.

Apparently this has been a bug for awhile but @ibuildthecloud did something to make it worse or make it happen more often. So, I'd like to keep this PR open until he fixes the bug so that we can retest after the fix.